File: 2020/02/day02_2.rb
lines = []
file = File.new("input.txt", "r")
while (line = file.gets)
  l = line.split
  password = {}
  password["min"], password["max"] = l[0].split("-")
  password["check"], password["pwd"] = l[1][0], l[2]
  lines.push(password)
end
file.close

valid_passwords = 0
lines.each do |line|
  if((line["pwd"][line["min"].to_i-1] == line["check"]) !=
    (line["pwd"][line["max"].to_i-1] == line["check"]))
    valid_passwords += 1
  end
end

puts(valid_passwords)

File: 2020/02/day02_2.go
package main

import (
  "bufio"
  "fmt"
  "log"
  "os"
  "strings"
  "strconv"
)

func main() {
  f, err := os.Open("input.txt")

  if err != nil {
    log.Fatal(err)
  }
  defer f.Close()

  var lines [][]string
  
  scanner := bufio.NewScanner(f)
  for scanner.Scan() {

    line := strings.Split(scanner.Text(), " ")  // spit the line
    tmp := strings.Split(line[0], "-")          // replace x-y with 2 individual elements
    _, line = line[0], line[1:]    
    line = append([]string{tmp[1]}, line...)
    line = append([]string{tmp[0]}, line...)
    line[2] = strings.TrimSuffix(line[2], ":")  // change x: into x
    
    lines = append(lines, line)
  }

  if err := scanner.Err(); err != nil {
    log.Fatal(err)
  }

  // check that only 1 of the places in string match
  valid_passwords := 0
  for _, i := range lines {    
    min, _ := strconv.Atoi(i[0])
    max, _ := strconv.Atoi(i[1])
        
    if (i[3][min-1:min] == i[2]) != (i[3][max-1:max] == i[2]) {
      valid_passwords++
    }

  }
  fmt.Println(valid_passwords)

}